### ISSUE TITLE
Potential Issue on data loader in distributed setting.

### ISSUE THREAD

**Issue by dingyuan-shi** [SalesforceAIResearch/DiffusionDPO#15]
Hello, 
Line 881 at train.py, it seems that the dataloader is not adapted for distributed setting.
The data entries will be repeatedly load and trained by different processes.
Maybe a sampler should be added, code as below:
```python
train_dataloader = torch.utils.data.DataLoader(
        train_dataset,
        shuffle=(args.split=='train'),
        collate_fn=collate_fn,
        batch_size=args.train_batch_size,
        num_workers=args.dataloader_num_workers,
        drop_last=True, 
        sampler=torch.utils.data.distributed.DistributedSampler(train_dataset), 
    )
```

**Comment by visionxyz** [comment]
I tried running it, but it seems that I need to remove `shuffle=(args.split=='train')` because it conflicts with `sampler`.

**Comment by avinabsaha** [comment]
@dingyuan-shi, could you share a sample configuration file to launch training across multiple nodes using "Accelerate Launch"?

My training is stuck after data loads into 1 GPU per node.

**Comment by visionxyz** [comment]
> @dingyuan-shi @visionxyz , could you share a sample configuration file to launch training across multiple nodes using "Accelerate Launch"?
> 
> My training is stuck after data loads into 1 GPU per node.

Hello, I just use 'accelerate launch --multi_gpu train.py', and make a config with 'accelerate config' first.

**Comment by avinabsaha** [comment]
Hi, @visionxyz  @dingyuan-shi  have you folks fine-tuned the SDXL model? I completed one iteration of SDXL fine-tuning and all the images coming out the model are very over-saturation. To verify, I also ran the models provided by the authors, and it seems to be working perfectly.
